File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.java
class Solution {
    public int solution(String t, String p) {
        int answer = 0;
        
        for(int i=0; i<t.length()-p.length()+1; i++) {
 
            String str = t.substring(i, i+p.length());
            long number1 = Long.parseLong(str);
            long number2 = Long.parseLong(p);

            if(number1 <= number2) {
                answer++;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

int solution(string t, string p) {
    int count = 0;
    
    for(int i=0; i<t.size()-p.size()+1; i++) {
        string sub = t.substr(i, p.size());

        long long number1 = stoll(sub);
        long long number2 = stoll(p);
        
        if(number1 <= number2) {
            count++;
        }
    }
    
    return count;
}